hub/src/server/drivers/AzDriver.ts

```diff
@@ -208,6 +208,75 @@ class AzDriver implements DriverModel, DriverModelTestMethods {
         ` ${azBlob} in container ${this.bucket}: ${error}`)
     }
   }
+
+  async performStat(args: PerformStatArgs): Promise<StatResult> {
+    if (!AzDriver.isPathValid(args.path)) {
+      throw new BadPathError('Invalid Path')
+    }
+    const azBlob = `${args.storageTopLevel}/${args.path}`
+    const blobURL = azure.BlobURL.fromContainerURL(this.container, azBlob)
+    const blockBlobURL = azure.BlockBlobURL.fromBlobURL(blobURL)
+    try {
+      const propertiesResult = await blockBlobURL.getProperties(azure.Aborter.none)
+      let lastModified: number | undefined
+      if (propertiesResult.lastModified) {
+        lastModified = Math.round(propertiesResult.lastModified.getTime() / 1000)
+      }
+      const result: StatResult = {
+        exists: true,
+        contentLength: propertiesResult.contentLength,
+        contentType: propertiesResult.contentType,
+        lastModifiedDate: lastModified
+      }
+      return result
+    } catch (error) {
+      if (error.statusCode === 404) {
+        const result: StatResult = {
+          exists: false
+        }
+        return result
+      }
+      /* istanbul ignore next */
+      logger.error(`failed to stat ${azBlob} in ${this.bucket}: ${error}`)
+      /* istanbul ignore next */
+      throw new Error('Azure storage failure: failed to stat' +
+        ` ${azBlob} in container ${this.bucket}: ${error}`)
+    }
+  }
+
+  async performRename(args: PerformRenameArgs): Promise<void> {
```

Comment:
I'm not saying that we need to address this just yet, but in all drivers that don't support a direct `rename`, including this one, there's a slim chance that the Gaia hub could crash while it's in the middle of a rename.  This would leak space -- it would leave behind a copy of the data that it meant to delete.  What are the security and correctness implications of this, considering that rename is used in collections?

Reply:
This should just result in an additional "archival" file that is a duplicate of the canonical file. The client initiating the request (blockstack.js or the browser) would end up with an http error or timeout, and any proceeding collection-related operations *should* be aborted and the whole operation would need retried at some other point. 

I don't think this would cause any other issues, will be good to keep in mind during further collection implementation though.